client/lib/url/resembles-url.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * Checks if the supplied string appears to be a URL.
+ * Looks only for the absolute basics:
+ *  - does it have a .suffix?
+ *  - does it have at least two parts separated by a dot?
+ *
+ * @param  query The string to check
+ * @return       Does it appear to be a URL?
+ */
+export default function resemblesUrl( query: string ): boolean {
+	if ( ! query ) {
+		return false;
+	}
+
+	let parsedUrl;
+	try {
+		parsedUrl = new URL( query );
+	} catch {
```

Comment:
Cool, TypeScript supports the empty catch, no need for `catch ( _unused_ ) {}` 👍